awx/main/tasks/facts.py

```diff
@@ -67,54 +73,74 @@ def start_fact_cache(hosts, destination, log_data, timeout=None, inventory_id=No
     msg='Inventory {inventory_id} host facts: updated {updated_ct}, cleared {cleared_ct}, unchanged {unmodified_ct}, took {delta:.3f} s',
     add_log_data=True,
 )
-def finish_fact_cache(hosts_cached, destination, facts_write_time, log_data, job_id=None, inventory_id=None):
+def finish_fact_cache(destination, log_data, job_id=None, inventory_id=None):
     log_data['inventory_id'] = inventory_id
     log_data['updated_ct'] = 0
     log_data['unmodified_ct'] = 0
     log_data['cleared_ct'] = 0
 
-    hosts_cached = sorted((h for h in hosts_cached if h.id is not None), key=lambda h: h.id)
+    summary_path = os.path.join(destination, 'host_cache_summary.json')
+    if not os.path.exists(summary_path):
+        system_tracking_logger.error(f'Missing summary file at {summary_path}')
+        return
+
+    try:
+        with open(summary_path, 'r', encoding='utf-8') as f:
+            summary = json.load(f)
+    except Exception:
+        system_tracking_logger.error(f'Unable to read summary file at {summary_path}')
+        return
+
+    host_names = summary.get('hosts_cached', [])
+    facts_write_time = os.path.getmtime(summary_path)
```

Comment:
This should work the same as before. My idea to solve the UTC problem was that we might use `os.path.getmtime` of _this_ file at `summary_path`. That way, if the remove node rewrites the modtime of the files, it does so consistently for the artifact files as well as this summary file.

Reply:
I believe I've addressed this comment now!